Fix 'Error Item' preserve recipes, fix all objects except coffee beans being treated as seeds

SinZ163 opened the pull request:
This fixes all objects (except Coffee Seeds) being treated as a Seed, along with fixing Error Preserves due to multiple (O) getting appended and becoming error items.

Before:
![image](https://github.com/Pathoschild/StardewMods/assets/767456/bcba4d38-4ff9-4586-ad79-6547444416d0)
![image](https://github.com/Pathoschild/StardewMods/assets/767456/26f48802-fda8-43f4-866e-e1bd4ec8494b)

After:
![image](https://github.com/Pathoschild/StardewMods/assets/767456/74ec52ec-0957-4caf-9cc2-cee125ea168f)
![image](https://github.com/Pathoschild/StardewMods/assets/767456/a9295d93-f9cf-415c-9172-020fc668fa40)

SinZ163 committed:
Fix Error Preserve recipes and fix everything (except Coffee) being a seed

Pathoschild commented on LookupAnything/DataParser.cs:
```diff
@@ -445,30 +445,26 @@ private Item CreateRecipeItem(string? inputID, string outputID, MachineRecipeOut
+                    switch (ItemRegistry.GetDataOrErrorItem(outputID).ItemId)
                     {
-                        case "(O)342":
                         case "342":
```
We should use the qualified item ID instead, to avoid conflicts with other `SObject`-type item IDs (e.g. furniture).

In that case we can also change `ItemRegistry.GetDataOrErrorItem(outputID).ItemId` to `ItemRegistry.QualifyItemId(outputId)`.

SinZ163 replied:
Done

Pathoschild commented on LookupAnything/DataParser.cs:
```diff
@@ -445,30 +445,26 @@ private Item CreateRecipeItem(string? inputID, string outputID, MachineRecipeOut
                     {
-                        case "(O)342":
                         case "342":
                             obj.preserve.Value = SObject.PreserveType.Pickle;
-                            obj.preservedParentSheetIndex.Value = inputID;
```
We can create flavored items using the game's new methods for it now, instead of doing it manually. For example:
```cs
return ItemRegistry
    .GetObjectTypeDefinition()
    .CreateFlavoredPickle(ItemRegistry.Create(inputId));
```

SinZ163 replied:
Done, though doesn't short circuit return to continue allow MachineRecipeOutputData to continue overriding if it exists

Pathoschild commented on LookupAnything/Framework/Lookups/Items/ItemSubject.cs:
```diff
@@ -467,10 +467,14 @@ private string GetTypeValue(Item item)
 
             try
             {
-                Crop crop = new(seed.ItemId, 0, 0, location ?? Game1.getFarm());
```
Similarly I suggest changing the condition above to avoid item ID collisions:
```cs
// current
if (seed is not SObject obj || obj.bigCraftable.Value)

// suggested
if (seed.HasTypeId(ItemRegistry.type_object))
```

SinZ163 replied:
Done (inverted HasTypeId to preserve the short circuit behavior)

Pathoschild reviewed (changes requested):
Thanks for the help! Just a few comments before it's merged.

SinZ163 committed:
Apply round 1 of PR feedback

Pathoschild committed:
avoid creating extra items when creating a recipe item

Pathoschild committed:
remove unneeded extra crop ID check in TryGetCropForSeed

The check will never fail in Stardew Valley 1.6, since an invalid seed will set the seed index to 0.

Pathoschild commented:
Merged into `develop` for the upcoming Lookup Anything 1.41.0. Thanks for the help!
